src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureCredentialsProvider.java

```diff
@@ -0,0 +1,139 @@
+            }
+            return list;
+        }
+
+        return Collections.emptyList();
+    }
+
+    @VisibleForTesting
+    static String generateKeyvaultItemName(String itemId) {
+        if (StringUtils.isEmpty(itemId)) {
+            throw new AzureKeyVaultException("Empty id for key vault item.");
+        }
+        int count = 0;
+        int index = -1;
+        for (int i = itemId.length() - 1; i >= 0; i--) {
+            if (itemId.charAt(i) == '/') {
+                count++;
+            }
+            if (count == 2) {
+                index = i;
+                break;
+            }
+        }
+        if (index < 0) {
+            throw new AzureKeyVaultException("Wrong pattern for key vault item id.");
+        }
+        return itemId.substring(index + 1);
+    }
+
+    private static Collection<IdCredentials> fetchCredentials() {
+        AzureKeyVaultGlobalConfiguration azureKeyVaultGlobalConfiguration = GlobalConfiguration.all().get(AzureKeyVaultGlobalConfiguration.class);
+        if (azureKeyVaultGlobalConfiguration == null) {
+            throw new AzureKeyVaultException("No global key vault url configured.");
+        }
+
+        String credentialID = azureKeyVaultGlobalConfiguration.getCredentialID();
+        KeyVaultCredentials keyVaultCredentials = AzureKeyVaultCredentialRetriever.getCredentialById(credentialID);
+        KeyVaultClient client = new KeyVaultClient(keyVaultCredentials);
+        String keyVaultURL = azureKeyVaultGlobalConfiguration.getKeyVaultURL();
+        List<IdCredentials> credentials = new ArrayList<>();
+        PagedList<SecretItem> secretItems = client.getSecrets(keyVaultURL);
+        for (SecretItem secretItem : secretItems) {
+            SecretBundle secret = client.getSecret(secretItem.id());
+            String id = secretItem.id();
+            IdCredentials cred = new SecretStringCredentials(CredentialsScope.GLOBAL, generateKeyvaultItemName(id),
+                    id, credentialID,
+                    secret.secretIdentifier().identifier());
+            credentials.add(cred);
+        }
+        PagedList<CertificateItem> certificateItems = client.getCertificates(keyVaultURL);
+        for (CertificateItem certificateItem : certificateItems) {
+            CertificateBundle certificate = client.getCertificate(certificateItem.id());
+            String id = certificateItem.id();
+            IdCredentials cred = new SecretCertificateCredentials(CredentialsScope.GLOBAL, generateKeyvaultItemName(id), id,
+                    credentialID,
+                    certificate.secretIdentifier().identifier(), Secret.decrypt(""));
+            credentials.add(cred);
+        }
+        client.httpClient().connectionPool().evictAll();
+        return credentials;
+    }
+
+    @Override
+    public CredentialsStore getStore(ModelObject object) {
+        return object == Jenkins.getInstance() ? store : null;
```

Comment:
this plugin uses semi recent jenkins base, use `Jenkins.get()` so as not to use deprecated code.
```suggestion
        return object == Jenkins.get() ? store : null;
```